### DISCUSSION TITLE
Disable `no-magic-numbers` for typed variables

### DISCUSSION THREAD

**Discussion by hmaesta** [typescript-eslint/typescript-eslint#9071] [Technical Discussions]
My `@typescript-eslint/no-magic-numbers` configuration allows `ignoreNumericLiteralTypes`, so I can create a type like the below without errors:

```tsx
type ElementMargin = 0 | 4 | 6 | 8 
```

Still, when using this type in a variable or a Map, for instance, I get "No magic number". If the variable is typed and its value is one of the expected literals, why would that be a magical number?

Some examples that return "No magic number":
```tsx
const margin: ElementMargin = isDesktop ? 6 : 4

const marginClassName: Map<ElementMargin, string> = new Map([
  [4, 'm-4'],
  [6, 'm-6'],
]);
```

🔗 [Playground link](https://typescript-eslint.io/play/#ts=5.4.3&fileType=.tsx&esQuery=N4IgZglgNgLgpgJxALhCAvkA&code=C4TwDgpgBAogNhAthAdsAsgQwE4HMCWKUAvFACxQA%2BUAbAFB0D0AVK84wwGYCuKAxsHwB7IgBlMIId2AAKAJRQA3nShQ%2BIgM7AoiHARQAuWAmRoseQiSj4NAEQgaA1sCFh5UAPy0oRsg1XYEMDc2EQyKqpQADzoIPBIqNq6FijEisn6AL6MAHwRcnSZDCxsHHTqKFo6eoQAwnCYGhoAcpjIRlhgUfGmGDUoADRQWtiEuDkkESgQAO5QnTIA2hGqi2RDAOSIALRkGwC6AytQizSbOzQHR6r7cgDcXLwCwkSxPYkyitUpUJlGXxlCEZ3mZ%2Br8FMoAkEQmFjlEACb4ABuagaTVayDSgJQmRyAAkIHA4EIhjMhNg4PCooxEUi8qoCkUmGx2FBHvxBCJrHYHM5XPIjAAjIRCBCYIiQqCBYKhKCcTBwDQQB6ZIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaAWwEMBzASwGNDdZiAjRaVDSfAM08gBpxskAAIAXAJ4AHFNWiUJIwiniVcIgPQESFGnQbNW6MAG0B2KC2j5ofU9ixnBlcgWiIAcgxY0AMpREtSeAAVSRRDEThEWwgAX1sAXQE4mKA&tsconfig=&tokens=false)

Is there any way to prevent "No magic number" for typed variables?

**Comment by bradzacher** [comment]
As per our contributing guide (which is listed in the template and you deleted) we do not use discussions for support or questions. Please read through the guide for alternatives.
https://typescript-eslint.io/contributing/discussions
